packages/ui/src/primitives/tailwind/Radio/index.tsx

```diff
@@ -25,82 +25,103 @@ Infinite Reality Engine. All Rights Reserved.
-      >
+}
+
+const RadioRoot = ({ defaultChecked, disabled, id, label, name, onChange, value, description }: RadioProps) => (
+  <div className="flex flex-col">
+    <div className="flex items-start gap-2">
+      <div className="mt-1 grid place-items-center">
         <input
           type="radio"
-          checked={selected}
-          value={value}
-          name={label}
-          onChange={onChange}
+          id={id}
+          name={name}
+          defaultChecked={defaultChecked}
           disabled={disabled}
-          className="shrink-0 rounded-full border-gray-200 text-blue-primary checked:border-blue-primary focus:ring-blue-primary disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 dark:border-gray-700 dark:bg-gray-800 dark:focus:ring-offset-gray-800"
+          className="
+            focus:ring-primary-blue
+            peer col-start-1
+            row-start-1 h-4
+            w-4 shrink-0 appearance-none rounded-full border-[1.5px]
+            border-blue-500 focus:outline-none focus:ring-[0.5px] focus:ring-offset-0
+            disabled:border-gray-400
+          "
+          onChange={onChange}
+          value={value}
         />
+        <div
+          className={cx(
+            'pointer-events-none',
+            'col-start-1 row-start-1',
+            'h-2 w-2 rounded-full peer-checked:bg-blue-500',
+            'peer-checked:peer-disabled:bg-gray-400'
+          )}
+        />
+      </div>
+      <label
+        htmlFor={id}
+        className={cx('text-start hover:cursor-pointer', {
+          'text-gray-400': disabled
+        })}
+      >
         {label}
       </label>
-      {description && <div className="ml-5 text-sm text-theme-primary">{description}</div>}
     </div>
-  )
+    {description && <div className="ml-6 text-sm text-gray-400">{description}</div>}
+  </div>
+)
+
+type OptionType = {
+  value: string
+  label: string
+  description?: string
+}
+export interface RadioGroupProps<T> {
+  defaultChecked?: string
+  value?: T
+  disabled?: boolean
+  name?: string
+  onChange: (value: T) => void
+  options: OptionType[]
+  horizontal?: boolean
+  className?: string
 }
 
 type OptionValueType = string | number
 
-const Radio = <T extends OptionValueType>({
-  value,
-  options,
-  onChange,
-  className,
-  labelClassName,
-  horizontal,
-  disabled
-}: {
-  value: T
-  options: { label: string; value: T; description?: string }[]
-  onChange: (value: T) => void
-  className?: string
-  labelClassName?: string
-  horizontal?: boolean
-  disabled?: boolean
-}) => {
+const Radio = <T extends OptionValueType>(props: RadioGroupProps<T>) => {
+  const { defaultChecked, disabled, name, onChange, options, horizontal, className } = props
+
+  const handleChange = (event: React.FormEvent<HTMLInputElement>) => onChange(event.currentTarget.value as T)
+
+  const defaultName = uuidv4()
```

Comment:
this will change of every render
we can just specify an empty string on Line#116

Reply:
good call fixed!